src/main/java/picard/sam/UmiMetrics.java

```diff
@@ -103,64 +87,27 @@ public UmiMetrics(final int length, final int observedUniqueUmis, final int infe
         INFERRED_UMI_ENTROPY = effectiveNumberOfBases(inferredUmis);
 
-        UMI_COLLISION_Q = -10 * Math.log10(UMI_COLLISION_EST / inferredUmis.size());
-        estimateBaseQualities(observedUmiBases);
+        UMI_BASE_QUALITIES = QualityUtil.getPhredScoreFromErrorProbability((double) OBSERVED_BASE_ERRORS / (double) observedUmiBases);
     }
 
     private double effectiveNumberOfBases(Histogram<?> observations) {
-        double entropyBase4 = 0.0;
-
         double totalObservations = observations.getSumOfValues();
-        for (Histogram.Bin observation : observations.values()) {
-            double pObservation = observation.getValue() / totalObservations;
-            entropyBase4 = entropyBase4 - pObservation * Math.log(pObservation);
-        }
 
         // Convert to log base 4 so that the entropy is now a measure
         // of the effective number of DNA bases.  If we used log(2.0)
         // our result would be in bits.
-        return entropyBase4 / Math.log(4.0);
-    }
-
-    public void updateDuplicateSetMetrics(final int maxEditDistanceToJoin, final int duplicateSetSize) {
-        DUPLICATE_SETS_WITH_UMI += duplicateSetSize;
-        DUPLICATE_SETS_WITHOUT_UMI++;
-
-        // For each duplicate set estimate the number of expected UMI collisions
-        double nWaysUmisCanDiffer = 0; // Number of ways two UMIs may contain errors, but be considered the same
-        for (int k = 0; k <= maxEditDistanceToJoin; k++) {
-            if (UMI_LENGTH > 0) {
-                nWaysUmisCanDiffer = nWaysUmisCanDiffer + LongMath.binomial(UMI_LENGTH, k) * Math.pow(3.0, k);
-            }
-        }
-
-        // The probability of two non-duplicate UMIs are drawn from a uniform distribution being correctly labeled as
-        // not belonging to the same UMI family is given as, pCorrectlyLabeled = nWaysUmisCanDiffer / 4^metrics.UMI_LENGTH.
-        // Estimate of probability all members in the duplicate set are correctly labeled is given by
-        // pAllMembersCorrectlyLabeled = (1 - pCorrectlyLabeled)^duplicateSets.size().
-        // The expected number of reads incorrectly labeled as belonging to a duplicate set is given as,
-        // (1 - pAllMembersCorrectlyLabeled) * duplicateSets.size().
-        UMI_COLLISION_EST = UMI_COLLISION_EST +
-                (1 - Math.pow(1 - nWaysUmisCanDiffer / Math.pow(4, UMI_LENGTH), duplicateSetSize - 1)) * duplicateSetSize;
-    }
-
-    public void updateUmiRecordMetrics(final String currentUmi, final String inferredUmi) {
-        OBSERVED_BASE_ERRORS += hammingDistance(currentUmi, inferredUmi);
-
-        observedUmiBases += UMI_LENGTH;
-        observedUmis.increment(currentUmi);
-        inferredUmis.increment(inferredUmi);
+        double entropyBase4 = observations.values().stream().collect(Collectors.summingDouble(
+                v -> -v.getValue() / totalObservations * Math.log(v.getValue() / totalObservations))) / Math.log(4.0);
```

Comment:
the function itself could be made clearer:
```
v->{ double p = v.getValue() / totalObservations;
return -p*log(p)/log(4.0); 
}
```
Also, if you need log base 4 a lot, perhaps you should make a LOG_BASE_4 object in MathUtil and use that. it will save on recomputing log(4.0) every time and, again make things clearer